tracer/src/Datadog.Trace/Telemetry/Collectors/MetricsTelemetryCollector.cs

```diff
@@ -119,7 +119,11 @@ public void Clear()
                                 metricName,
                                 points: new MetricSeries { new(timestamp, value) },
                                 common: metric.IsCommon(),
-                                type: TelemetryMetricType.Count) { Tags = metricKey.Tags });
+                                type: TelemetryMetricType.Count)
+                            {
+                                Namespace = metric.GetNamespace(),
```

Comment:
I think you were right, and maybe having snapshots would help us catch these more easily. WDYT?

Reply:
Absolutely, I'm a big fan of snapshots obviously, my main question/concern is _where_ do we snapshot? Ideally we want to snapshot the payloads we send in integration tests, but those metrics payloads are somewhat variable depending on exactly when they get sent... I think there's still something we could/should do there though